Ribbon review

Blake-Madden opened the pull request:
Various small refinements to ribbon code

Blake-Madden committed:
Fix wrong colors being copied and returned

Blake-Madden committed:
Add bitmap and pointer validations

Blake-Madden committed:
Add consistent use of wxDynamicCast

Blake-Madden committed:
Fix wrong flag being toggled in toolbar

Blake-Madden committed:
Fix DeletePage when zero is given
Would then try to call SetActivePage with max size_t. Code is now simplified and stays within boundary of the number of pages.

paulcor commented on src/ribbon/gallery.cpp:
```diff
@@ -543,7 +544,7 @@ wxRibbonGalleryItem* wxRibbonGallery::Append(const wxBitmap& bitmap, int id)
     wxRibbonGalleryItem *item = new wxRibbonGalleryItem;
     item->SetId(id);
-    item->SetBitmap(bitmap);
+    item->SetBitmap(bitmap.IsOk() ? bitmap : wxBitmap{});
```
This doesn't do anything. "Not ok" is the same as "empty".

Blake-Madden replied:
Rolled that back.

Blake-Madden committed:
Undo bitmap check that didn't serve a purpose

vadz reviewed (commented):
Thanks, most of the changes look good to me but I'm not sure about 62384e2da56a48c2d757094aff08e733d04ac12d: why should we use `wxDynamicCast()` here if we don't check its result anyhow? If the parent is always `wxRibbonBar`, as I think it should be, then `wxStaticCast()` is better if only because it's more clear about the fact that it is always supposed to succeed.

Blake-Madden commented:
> why should we use `wxDynamicCast()` here

`wxDynamicCast(GetParent(), wxRibbonBar)` is used every else in these files, but without the post-cast pointer check also. I would be happy to add pointer checks to all of these. I tend to be overly pedantic and dynamic cast would be my preference (with pointer check) to absolutely be sure things are what we think they are when casting and using them.

If you prefer static cast, I should change that elsewhere too to make things more consistent in these files.

Blake-Madden committed:
When dynamic cast was used and pointer wasn't check, replace with DEBUG type check and static_cast
Other existing dynamic casts where return pointer was validated is left in place.

Blake-Madden committed:
Add pointer check to suppress PREFast null deref warning

Blake-Madden committed:
Use wxASSERT_MSG where needed

Blake-Madden committed:
Add and use new wxCheckedStaticCast helper

vadz commented:
Looks good, thanks! Will merge soon.
